src/NCPopupMenu.cc

```diff
@@ -141,3 +171,79 @@ bool NCPopupMenu::postAgain()
     return again;
 }
 
+
+NCursesEvent NCPopupMenu::select_next_item()
+{
+    NCursesEvent event;
+
+    YTableItem * tableItem = dynamic_cast<YTableItem *> ( getCurrentItemPointer() );
+
+    if ( !tableItem)
+	return event;
+
+    auto current = find_item(tableItem);
+
+    if ( current == _items.end() )
+	return event;
+
+    auto begin = std::next(current, 1);
+
+    auto next_enabled = find_if(begin, _items.end(), [](const Item * item) {
```

Comment:
Please move that whole obscure check out to a separate function with a speaking name. A casual reader has no chance to figure out what is going on here.